lib/meadow/pipeline/actions/extract_exif_metadata.ex

```diff
@@ -0,0 +1,57 @@
+defmodule Meadow.Pipeline.Actions.ExtractExifMetadata do
+  @moduledoc """
+  Action to extract the EXIF metadata from a FileSet
+
+  Subscribes to:
+  * Kicked off manually
+
+  """
+  alias Meadow.Config
+  alias Meadow.Data.{ActionStates, FileSets}
+  alias Meadow.Utils.Lambda
+  alias Sequins.Pipeline.Action
+
+  use Action
+  use Meadow.Pipeline.Actions.Common
+
+  require Logger
+
+  @actiondoc "Extract EXIF metadata from FileSet"
+
+  defp process(%{file_set_id: file_set_id}, _attributes, _) do
+    file_set = FileSets.get_file_set!(file_set_id)
+    ActionStates.set_state!(file_set, __MODULE__, "started")
+
+    try do
+      Logger.info("Extracting EXIF metadata from #{file_set.id}")
+
+      exif_metadata = extract_exif_metadata(file_set.metadata.location)
+      Logger.info("EXIF metadata extracted from #{file_set.id}")
+
+      {:ok, _} =
+        file_set
+        |> FileSets.update_file_set(%{metadata: %{exif: exif_metadata}})
+
+      ActionStates.set_state!(file_set, __MODULE__, "ok")
+      :ok
+    rescue
+      e ->
+        ActionStates.set_state!(file_set, __MODULE__, "error", Exception.message(e))
+        {:error, Exception.message(e)}
+    end
+  end
+
+  def extract_exif_metadata(url) do
+    case URI.parse(url) do
+      %{host: bucket, path: "/" <> key} -> invoke_lambda(bucket, key)
+      _ -> Logger.error("Cannot extract EXIF from invalid url: #{url}")
+    end
+  end
+
+  defp invoke_lambda(bucket, key) do
+    case Lambda.invoke(Config.lambda_config(:exif), %{bucket: bucket, key: key}) do
+      {:ok, result} -> result
```

Comment:
```
iex(13)> Lambda.invoke(Config.lambda_config(:exif), %{bucket: bucket, key: key})
[info]  Extracting EXIF metadata: s3://dev-preservation//6b/13/6e/fd/a5acb8dc829138cc0032d0e38a57cebaabe2bf5d90239127eae3706e96b7dfb9
[warn]  Received undefined response from lambda
{:ok, nil}
```


So, our seed data (jpgs) seem to return `undefined` from `exifr`, which `lambda.ex` logs and turns into `{:ok, nil}`. It seems like we're catching the nil in that pattern match as a valid result and going ahead with an attempted db update. Can we change that so it recognizes the undefined/nil?

I'm a little confused why `exifr` returns undefined. With `exiftool` we seem to at least get image width, height, mime type, etc...

Reply:
@kdid Just pushed up a commit so that case is handled explicitly. I totally agree about the seed data issue, but I think what is happening is that `exifr` is constrained to just pulling the `EXIF` and `JFIF` metadata from the files, whereas `exiftool` does some extra work to pull things like height and width even if they are not explicitly set in the metadata. In the interest of moving towards go-live I think we should probably just move ahead with what we've got and circle back to this once we have some more time.

Reply:
Yeah, agree, I didn't mean that you should fix that just the handling of the `nil` response...